File: 100-doors/C#/100-Doors/DoorKata.cs
using System;

namespace DoorKataLib
{
    public class DoorKata
    {
        public void RunDoorKata(bool[] doors)
        {
            for (int pass = 1; pass <= doors.Length; pass++)
            {
                for (int door = pass - 1; door < doors.Length; door += pass)
                {
                    doors[door] = !doors[door];
                }
            }
        }

        public string GetDoorStateString(bool[] doors)
        {
            char[] doorChars = new char[doors.Length];
            for (int i = 0; i < doors.Length; i++)
            {
                doorChars[i] = doors[i] ? '@' : '#';
            }
            return new string(doorChars);
        }
    }
}

File: 100-doors/python/door_kata.py
class DoorKata:
    def run_door_kata(self, doors):
        for pass_num in range(1, len(doors) + 1):
            for door in range(pass_num - 1, len(doors), pass_num):
                doors[door] = not doors[door]

    def get_door_state_string(self, doors):
        return ''.join('@' if door else '#' for door in doors)

